lib/pure/parsecfg.nim

```diff
@@ -482,54 +494,62 @@ proc replace(s: string): string =
     inc(i)
   result = d
 
+proc writeConfigToStream*(dict: Config, stream: Stream) =
+  ## Writes the contents of the table to the specified stream
+  ## Note: Comment statement will be ignored.
+  var section, key, value, kv, segmentChar:string
+  for pair in dict.pairs():
+    section = pair[0]
+    if section != "": ## Not general section
+      if not allCharsInSet(section, SymChars): ## Non system character
+        stream.writeLine("[\"" & section & "\"]")
+      else:
+        stream.writeLine("[" & section & "]")
+    for pair2 in pair[1].pairs():
+      key = pair2[0]
+      value = pair2[1]
+      if key[0] == '-' and key[1] == '-': ## If it is a command key
+        segmentChar = ":"
+        if not allCharsInSet(key[2..key.len()-1], SymChars):
+          kv.add("--\"")
+          kv.add(key[2..key.len()-1])
+          kv.add("\"")
+        else:
+          kv = key
+      else:
+        segmentChar = "="
+        kv = key
+      if value != "": ## If the key is not empty
+        if not allCharsInSet(value, SymChars):
+          if find(value, '"') == -1:
+            kv.add(segmentChar)
+            kv.add("\"")
+            kv.add(replace(value))
+            kv.add("\"")
+          else:
+            kv.add(segmentChar)
+            kv.add("\"\"\"")
+            kv.add(replace(value))
+            kv.add("\"\"\"")
+        else:
+          kv.add(segmentChar)
+          kv.add(value)
+      stream.writeLine(kv)
+
+proc writeConfigToString*(dict: Config): string =
```

Comment:
Hrm. I dislike this name as well.

I think `$` is the way to go here. Mainly because ``JsonNode.$`` has this behaviour too.